### ISSUE TITLE
SymbolicRegression compatibility issues

### ISSUE THREAD

**Issue by lgrozinger** [MilesCranmer/SymbolicRegression.jl#32]
`SymbolicRegression` lists a compatibility requirement for `SymbolicUtils` of `0.6`. This makes it incompatible with other packages that require newer versions of `SymbolicUtils`. E.g. `Symbolics`, which seems like a useful package to use in conjunction with `SymbolicRegression`, but causes problems with installation as in the attempt below:

```
(@v1.6) pkg> add https://github.com/MilesCranmer/SymbolicRegression.jl
     Cloning git-repo `https://github.com/MilesCranmer/SymbolicRegression.jl`
    Updating git-repo `https://github.com/MilesCranmer/SymbolicRegression.jl`
   Resolving package versions...
ERROR: Unsatisfiable requirements detected for package SymbolicUtils [d1185830]:
 SymbolicUtils [d1185830] log:
 ├─possible versions are: 0.1.0-0.12.0 or uninstalled
 ├─restricted to versions * by ElectroBiofilm [ee21f0d5], leaving only versions 0.1.0-0.12.0
 │ └─ElectroBiofilm [ee21f0d5] log:
 │   ├─possible versions are: 0.1.0 or uninstalled
 │   └─ElectroBiofilm [ee21f0d5] is fixed to version 0.1.0
 ├─restricted to versions 0.6 by SymbolicRegression [8254be44], leaving only versions 0.6.0-0.6.3
 │ └─SymbolicRegression [8254be44] log:
 │   ├─possible versions are: 0.6.9 or uninstalled
 │   └─SymbolicRegression [8254be44] is fixed to version 0.6.9
 └─restricted by compatibility requirements with Symbolics [0c5d862f] to versions: 0.8.4-0.12.0 — no versions left
   └─Symbolics [0c5d862f] log:
     ├─possible versions are: 0.1.0-1.1.0 or uninstalled
     └─restricted to versions * by ElectroBiofilm [ee21f0d5], leaving only versions 0.1.0-1.1.0
       └─ElectroBiofilm [ee21f0d5] log: see above
```

Is there any way to relax the dependency on the old version of `SymbolicUtils`? Is it a matter of some superficial changes to accomadate changes in the `SymbolicUtils` API, or does it really truly depend on `0.6.x`?

**Comment by MilesCranmer** [comment]
Unfortunately it truly depends on only 0.6.x. See https://github.com/MilesCranmer/SymbolicRegression.jl/issues/24#issuecomment-846302475 for more discussion about this. Would be nice to have this working in the future, but it will take quite a bit of work to update it since their whole backend API which I use changed. 

If curious, these are the files which need to be updated: https://github.com/MilesCranmer/SymbolicRegression.jl/blob/master/src/CustomSymbolicUtilsSimplification.jl and https://github.com/MilesCranmer/SymbolicRegression.jl/blob/master/src/InterfaceSymbolicUtils.jl.

**Comment by AlCap23** [comment]
#37 Should handle this

**Comment by MilesCranmer** [comment]
Fixed by @AlCap23 in v0.6.13 onwards
